collector/meminfo_bsd.go

```diff
@@ -22,30 +22,66 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+type sysctlType uint8
+
+const (
+	t_uint32 sysctlType = iota
+	t_uint64
+)
+
+type meminfoSysctl struct {
+	name       string
+	dataType   sysctlType
+	conversion func(uint64) uint64
+}
+
 func (c *meminfoCollector) getMemInfo() (map[string]float64, error) {
+	var tmp32 uint32
+	var tmp64 uint64
+	var err error
+
 	info := make(map[string]float64)
 
-	size, err := unix.SysctlUint32("vm.stats.vm.v_page_size")
+	tmp32, err = unix.SysctlUint32("vm.stats.vm.v_page_size")
```

Comment:
Trying to understand what you are getting at here.  My thinking is that I want to cast to uint64 outside of the closure, and I'm pretty sure I'll get a type mis-match without a cast.  Am I missing something?o